### PR TITLE
Week 2 nico perrault

### PR DESCRIPTION
**Pull request by nouxcode**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [  ✅ ] Coding Challenges
- [ ✅  ] Coding Concepts
- [  ✅ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -32,28 +45,133 @@ const object2 = { number: 0 }
...
+  
+
+  const capitalizedArray = (array) => {
+    return array.map((value) => {
+      return value.charAt(0).toUpperCase() + value.slice(1).toLowerCase()
+    })
+  }
+
+//*Test Suites: 1 passed, 1 total
+// The test passed, all systems go :)
```
Comment (elysemontano) [inline code-challenges.test.js:122]: 💯

File: code-challenges.test.js [hunk 2]
```diff
@@ -32,28 +45,133 @@ const object2 = { number: 0 }
...
+      const vowelTester1 = "learn"
+      // Expected output: 1
+      const vowelTester2 = "academy"
+      // Expected output: 0
+      const vowelTester3 = "challenges"
+      // Expected output: 2
+      expect(firstVowel(vowelTester1)).toEqual(1)
+      expect(firstVowel(vowelTester2)).toEqual(0)
+      expect(firstVowel(vowelTester3)).toEqual(2)
+      expect(firstVowel("glyph")).toEqual("glyph does not contain a vowel.")
```
Comment (elysemontano) [inline code-challenges.test.js:141]: Nice work adding an edge case in here!

File: coding-concepts.js [hunk 3]
```diff
@@ -20,24 +21,27 @@ const greeter = (name) => {
 }
 // console.log(greeter("LEARN Student"))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: I think that this will log the output Hello, LEARN Student!
+
+// b) Verify and explain: My answer was correct because the string used string interperlation as it was passed within the function.
```
Comment (elysemontano) [inline coding-concepts.js:26]: This should have returned undefined since there is no return inside the function.

File: coding-concepts.js [hunk 4]
```diff
@@ -10,8 +10,9 @@
 const cohort = "Foxtrot 2022"
 // console.log(cohort.split(""))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: I think that this will log the output ['F', 'o', 'x', 't', 'r', 'o', 't', ' ', '2', '0', '2', '2']
```
Comment (elysemontano) [inline coding-concepts.js:13]: Awesome work in this section!

File: interview-questions.md [hunk 5]
```diff
@@ -8,50 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: They both return a new array of elements where you have applied some function on the element so that it changes the original element typically.
 
 4. What is iteration?
 
-Your answer:
+Your answer: An iteration is a way of repeating something easily within coding.
 
-Researched answer:
+Researched answer: Iteration is a class of array methods.  Iteration methods operate on every item in an array, one at a time.
```
Comment (elysemontano) [inline interview-questions.md:31]: You can actually iterate on both strings and arrays when using a for loop.

**Review by elysemontano — APPROVED** [review]
Great work Nico!  I really appreciate how thorough your pseudocode is.  It was also really great to see you adding edge cases into your tests.  Keep up the awesome work!
